nilearn/plotting/displays.py

```diff
@@ -350,6 +350,120 @@ def _add_lines(self, line_coords, line_values, cmap,
             self.ax.add_line(line)
 
 
+class GlassBrainHemisphericAxes(GlassBrainAxes):
+    """An MPL axis-like object that displays a 2D projection of 3D
+    volumes with a schematic view of the brain. In the case of saggital cuts,
+    projects only data in the selected left or right hemisphere.
+
+    """
+    def __init__(self, ax, direction, coord, plot_abs=True, **kwargs):
+        super(GlassBrainHemisphericAxes, self).__init__(ax, direction, coord)
+        self._plot_abs = plot_abs
+        if ax is not None:
+            object_bounds = glass_brain.plot_brain_schematics(ax,
+                                                              direction,
+                                                              **kwargs)
+            self.add_object_bounds(object_bounds)
+
+    def draw_2d(self, data_2d, data_bounds, bounding_box,
+                type='imshow', **kwargs):
+        # kwargs massaging
+        kwargs['origin'] = 'upper'
+
+        if self.direction == 'y':
+            (xmin, xmax), (_, _), (zmin, zmax) = data_bounds
+            (xmin_, xmax_), (_, _), (zmin_, zmax_) = bounding_box
+        elif self.direction in 'lr':
+            (_, _), (xmin, xmax), (zmin, zmax) = data_bounds
+            (_, _), (xmin_, xmax_), (zmin_, zmax_) = bounding_box
+        elif self.direction == 'z':
+            (xmin, xmax), (zmin, zmax), (_, _) = data_bounds
+            (xmin_, xmax_), (zmin_, zmax_), (_, _) = bounding_box
+        else:
+            raise ValueError('Invalid value for direction %s' %
+                             self.direction)
+        ax = self.ax
+        # Here we need to do a copy to avoid having the image changing as
+        # we change the data
+        im = getattr(ax, type)(data_2d.copy(),
+                               extent=(xmin, xmax, zmin, zmax),
+                               **kwargs)
+
+        self.add_object_bounds((xmin_, xmax_, zmin_, zmax_))
+
+        return im
+
+    def draw_left_right(self, size, bg_color, **kwargs):
+        if self.direction in 'lr':
+            return
+        ax = self.ax
+        ax.text(.1, .95, 'L',
+                transform=ax.transAxes,
+                horizontalalignment='left',
+                verticalalignment='top',
+                size=size,
+                bbox=dict(boxstyle="square,pad=0",
+                          ec=bg_color, fc=bg_color, alpha=1),
+                **kwargs)
+
+        ax.text(.9, .95, 'R',
+                transform=ax.transAxes,
+                horizontalalignment='right',
+                verticalalignment='top',
+                size=size,
+                bbox=dict(boxstyle="square,pad=0", ec=bg_color, fc=bg_color),
+                **kwargs)
+
+    def transform_to_2d(self, data, affine):
```

Comment:
I also noticed that some changes would just mean one line in the parent classes. Nonetheless I put all the changes in the GlassBrainHemisphericAxes class for 2 reasons. First its not just GlassBrainAxes but also BaseAxes that would have modifications, I thought having spread changes across multiple classes for one feature might be confusing in the future and could break some other derived class (so far its assume for all the plotting framework that only xyz exist). Second I am not sure how the changes introduced would interact with the connectomics plots that also use the GlassBrainAxes class. I think putting this together and checking how this should be used with the connectomics should be a different PR. @GaelVaroquaux what do you think?